### PR TITLE
Vertical stepper implemenetation

### PR DESCRIPTION
**Pull request by ashish-egov**
## Release Notes

- **New Features**
  - Introduced `UploadDataCustom` component for handling data file uploads with validation and feedback mechanisms.
  - Implemented `XlsPreview` component for displaying Excel file previews in a user-friendly popup.
  - New configuration step for data uploading in the `MicroplanConfig`, enhancing user workflow.

- **Improvements**
  - Enhanced download functionality to allow custom file naming during Excel downloads.
  - Integrated event listener in `SetupMicroplan` for dynamic URL parameter handling.
  - Improved code readability and consistency across various components, including formatting adjustments.

- **Bug Fixes**
  - Adjusted component registration and step numbering in the Microplan module for better clarity and functionality.

### CODE REVIEW

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/hooks/useGenerateIdCampaign.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export const useGenerateIdCampaign = ({ type, hierarchyType, filters, campaignId, config = {} }) => {
+export const useGenerateIdCampaign = ({ type, hierarchyType, filters, campaignId,source=null, config = {} }) => {
```
[the remark this answers is not in the document]
Comment (ashish-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/hooks/useGenerateIdCampaign.js:1]: auto assignment is needed

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/SetupMicroplan.js [hunk 2]
```diff
@@ -94,6 +94,22 @@ const SetupMicroplan = ({ hierarchyType, hierarchyData }) => {
     setMicroplanConfig(MicroplanConfig(params, null, isSubmitting, null, hierarchyData));
   }, [totalFormData, isSubmitting]);
 
+  useEffect(() => {
+    const handleCheckingEvent = () => {
+      console.log("Checking event triggered");
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/SetupMicroplan.js:99]: remove logs

File: health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/utils/downloadExcel.js [hunk 3]
```diff
@@ -23,19 +23,25 @@ export const downloadExcelWithCustomName = ({ fileStoreId = null, customName = n
...
-          "Content-Type": "application/json",
-          Accept: "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
-          "auth-token": Digit.UserService.getUser()?.["access_token"],
-        },
-        params: {
-          tenantId: Digit.ULBService.getCurrentTenantId(),
-          fileStoreId: fileStoreId,
-        },
-      })
+    axios.request({
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/utils/downloadExcel.js:26]: test if this change in required @ashish-egov @Bhavya-egov

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/XlsPreview.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import { PopUp, SVG, DownloadIcon, Button } from "@egovernments/digit-ui-react-components";
+import React from "react";
+import DocViewer, { DocViewerRenderers } from "@cyntler/react-doc-viewer";
+import { useTranslation } from "react-i18next";
+import { PRIMARY_COLOR } from "../utils";
+
+const ArrowBack = ({ className = "", height = "15", width = "15", styles = {} }) => {
+  return (
+    <svg className={className} style={styles} width={width} height={height} viewBox="0 0 15 14" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/XlsPreview.js:9]: Re use this from campaign

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/SetupMicroplanConfig.js [hunk 5]
```diff
@@ -147,59 +147,87 @@ export const MicroplanConfig = (totalFormData, dataParams, isSubmitting, summary
             },
           ],
         },
-       
         {
-          stepCount: "7",
+          stepCount: "5",
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/SetupMicroplanConfig.js:151]: update this stepcount according to the screen sequence in figma

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/SetupMicroplan.js [hunk 6]
```diff
@@ -94,6 +94,21 @@ const SetupMicroplan = ({ hierarchyType, hierarchyData }) => {
     setMicroplanConfig(MicroplanConfig(params, null, isSubmitting, null, hierarchyData));
   }, [totalFormData, isSubmitting]);
 
+  useEffect(() => {
+    const handleCheckingEvent = () => {
+      const newKey = parseInt(new URLSearchParams(window.location.search).get("key")) || 1;
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/SetupMicroplan.js:99]: Use Digit.hooks.useQueryParams

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/SetupMicroplan.js [hunk 7]
```diff
@@ -195,43 +208,51 @@ const SetupMicroplan = ({ hierarchyType, hierarchyData }) => {
...
+
+
+
+  };
 
   if (isLoadingCampaignObject || isLoadingPlanObject) {
     return <Loader />;
   }
 
+  var aa = filteredConfig?.[0]?.form.map((config) => {
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/SetupMicroplan.js:248]: remove this

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/utils/downloadExcel.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import axios from "axios";
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/utils/downloadExcel.js:1]: reuse from campaign

File: health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/UploadData.js [hunk 9]
```diff
@@ -1040,93 +1040,93 @@ const UploadData = ({ formData, onSelect, ...props }) => {
   return (
     <>
       <div className="container-full">
-      <div className="card-container">
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/UploadData.js:1043]: Are these updates to campaign-manager/ required?

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**